handlers.go

```diff
@@ -3153,6 +3155,66 @@ func (s *Server) handleAddContentsToCollection(c echo.Context, u *User) error {
 	return c.JSON(200, map[string]string{})
 }
 
+// handleCommitCollection godoc
+// @Summary      Produce a CID of the collection contents
+// @Description  This endpoint is used to save the contents in a collection, producing a top-level CID that references all the current CIDs in the collection.
+// @Param        coluuid     path     string  true     "coluuid"
+// @Tags         collections
+// @Produce      json
+// @Success      200  {object}  string
+// @Router       /collections/commit/{collection-id} [post]
+func (s *Server) handleCommitCollection(c echo.Context, u *User) error {
+	colid := c.Param("coluuid")
+
+	var col Collection
+	if err := s.DB.First(&col, "uuid = ? and user_id = ?", colid, u.ID).Error; err != nil {
+		return err
+	}
+
+	contents := []Content{}
+	if err := s.DB.Debug().
+		Model(CollectionRef{}).
+		Where("collection = ?", col.ID).
+		Joins("left join contents on contents.id = collection_refs.content").
+		Select("contents.*").
+		Scan(&contents).Error; err != nil {
+		return err
+	}
+
+	bserv := blockservice.New(s.Node.Blockstore, offline.Exchange(s.Node.Blockstore))
+	dserv := merkledag.NewDAGService(bserv)
+
+	collectionNode := merkledag.ProtoNode{}
+	for _, content := range contents {
+		curCID := content.Cid.CID
+		nd, err := dserv.Get(context.Background(), curCID)
```

Comment:
I don't think this is the best way to do this;

1 - How would you get blocks for content hosted in shuttles?
2 - They can be a simpler way, using an empty dir node 😸 

```
// create a dir to wrap the collection content at this point in time - everytime the collection content changes, this dir cid/commit changes
	collectionNode := unixfs.EmptyDirNode()
	for _, c := range contents {
		collectionNode.AddRawLink(c.Name, &ipld.Link{
			Size: uint64(c.Size),
			Cid:  c.Cid.CID,
		})
	}